packages/cmf/src/reducers/componentsReducers.js

```diff
@@ -14,17 +15,31 @@ export const defaultState = new Map();
 export function componentsReducers(state = defaultState, action) {
 	switch (action.type) {
 	case ACTIONS.componentsActions.COMPONENT_ADD_STATE:
+		if (state.getIn([action.componentName, action.key])) {
+			invariant(
+				false,
+				`Can't set up your component ${action.componentName} on
+				key ${action.key} since this association already exist`,
+			);
+		}
 		if (action.initialComponentState) {
 			return state.setIn(
 				[action.componentName, action.key],
 				fromJS(action.initialComponentState)
 			);
 		}
+
 		return state.setIn(
 			[action.componentName, action.key],
 			new Map()
 		);
 	case ACTIONS.componentsActions.COMPONENT_MERGE_STATE:
+		if (!state.getIn([action.componentName, action.key])) {
+			let msg = 'The component state can\'t be merged since the ';
+			msg += `${action.componentName}, ${action.key} association doesn't exist.`;
```

Comment:
Template string for the entire string sound's better than concatenation. Less assignation ;)